bciers/apps/reporting/src/app/components/facility/FacilityEmissionAllocationForm.tsx

```diff
@@ -160,13 +159,21 @@ export default function FacilityEmissionAllocationForm({
         .filter((category: any) => category.category_type === "fuel_excluded")
         .map(calculateEmissionData),
     total_emission_allocations: {
-      facility_total_emissions:
-        initialData.facility_total_emissions?.toString(),
-      products: initialData.report_product_emission_allocation_totals,
+      facility_total_emissions: initialData.facility_total_emissions
+        ? Number(initialData.facility_total_emissions)
+        : initialData.facility_total_emissions,
+      products:
+        initialData.report_product_emission_allocation_totals?.map(
+          (product: { [key: string]: any }) => ({
+            ...product,
+            allocated_quantity: product.allocated_quantity
+              ? Number(product.allocated_quantity)
+              : product.allocated_quantity,
+          }),
```

Comment:
I think the logic here could be put behind a function? is the idea to "parse number or undefined" ?
We might be able to handle things a bit better by having the code be explicit about what is expected. Otherwise empty string, empty array, null or undefined would all stay unchanged